pandas/core/internals/blocks.py

```diff
@@ -671,15 +671,6 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
                 raise
             newb = self.copy() if copy else self
 
-        if newb.is_numeric and self.is_numeric:
-            if newb.shape != self.shape:
```

Comment:
Does anyone know what case meets the condition?

Reply:
I *think* .astype('i1') on an int64 might hit this

Reply:
Thx. It looks work (might depending on OS?)

```
pd.DataFrame([1]).astype('i1')
```

To keep the condition for safety, remove itemsize from the error message? (or add `.itemsize` to extension dtypes)

```
pd.Series([1], dtype='int64').dtype.itemsize
# 8
pd.Series([1], dtype='Int64').dtype.itemsize
# AttributeError: 'Int64Dtype' object has no attribute 'itemsize'
```